client/client.go

```diff
@@ -25,13 +26,13 @@ const (
 var httpReqError = errors.New("error building http request")
 
 // Returns a custom AdminPortal which integrates with the users Account Management API.
-// Supported schemes are http and https
-func NewAdminPortal(scheme string, host string, port int) (*AdminPortal, error) {
-	url2, err := verifyUrl(fmt.Sprintf("%s://%s:%d", scheme, host, port))
+func NewAdminPortal(systemURL string) (*AdminPortal, error) {
+	parsed, err := url.ParseRequestURI(systemURL)
 	if err != nil {
 		return nil, err
 	}
-	return &AdminPortal{scheme, host, port, url2}, nil
+	systemURL = strings.TrimSuffix(systemURL, "/")
```

Comment:
you can achieve the same calling `parsed.Host`, right?

A little thing also: I would not use `systemURL` term. System is internal 3scale name. `adminPortalURLStr` or `portaURLStr` seems more appropriate.

Reply:
No I think `parsed.Host` will strip the path which we want to preserve to fix #31

Reply:
Sorry, I do not get it. Why would you like to preserve the path? The full path is being defined by the operation like `get account list` -> `/admin/api/accounts.json`

Reply:
There is a use case in which we want to support a root path for the API that is not `/`, but something else, such as having the API mounted in `/threescale-api/`, so that the operation would end up with `GET /threescale-api/admin/api/accounts.json`.

Reply:
All endpoints of the *Master API* are prefixed with `/master`. Some of them are already available in this client, like for instance
```
tenantCreate = "/master/api/providers.json"
```

Prefixing all endpoints with a fix prefix, like `master`,  would break the *Master API* endpoints and others. I do not think it is available limits endpoints prefixed with master: `/master/admin/api/application_plans/%d/limits.json`.

Checking 3scale API docs for the master account, all endpoints in the *Account Management API` are prefixed with `/admin/api` and all endpoints in the `Master API` are prefixed with `/master/api`. I cannot find any endpoint like `/master/admin/api/accounts.json`. This is not surprising, though, as `/master/api/proxy/configs/production.json` endpoint is being used by apicast to fetch all services from all accounts and it is not documented.

My point is that I am not sure about the convenience of prefixing all the endpoints with a fixed prefix. Maybe it would be better to add a new operation like `Get master account list`
```
masterAccountList = "/master/admin/api/accounts.json"
```

The change does not break backward compatibility and I am not blocking it. I am just asking you to have a though about this. 

If still you want to add the new constructor with a fixed prefix, I will approve it. 

Let me know what you think.

Reply:
@eguzki I think @unleashed point is that someone may install porta under a specific path and we should support that. `master` was the example that was given to us but it could be anything.

I do agree with you that I find it unusual that someone would want to do this but I guess there is no harm in supporting such an installation.